syntax = "proto3";

package transport_catalogue_serialize;

message Coordinates {
    double latitude = 1;                               
    double longitude = 2;                              
}

message Stop {
    string stop_name = 1;                               
    Coordinates stop_coordinates = 2;                   
}

message Bus {
    string bus_name = 1;                                
    repeated string bus_stops = 2;                      
    uint64 unique_stops_qty = 3;                           
	double geo_route_length = 4;                          
	uint64 real_route_length = 5;                          
	double curvature = 6;                                  
	bool is_circular = 7;                                  
}

message Distance {
    string from = 1;                                     
    string to = 2;                                       
    uint64 range = 3;                                   
}

import public "map_renderer.proto";

message RouterSettings {
    uint64 bus_wait_time = 1;                             
    double bus_velocity = 2;                               
}

import public "transport_router.proto";

message TransportCatalogue {
	repeated Stop stops_data = 1;                        
	repeated Bus buses_data = 2;                         
    repeated Distance distances_data = 3;                
    RendererSettings renderer_settings = 4;              
    RouterSettings router_settings = 5;                  
    RouterData router_data = 6;
}